libraries/chain_kv/include/b1/session/shared_bytes.hpp

```diff
@@ -508,8 +508,14 @@ shared_bytes::shared_bytes_iterator<Iterator_traits>::operator[](size_t index) c
    return m_buffer[index];
 }
 
+struct shared_bytes_reflect{
```

Comment:
So, instead of making this struct and reflecting it.  What you should do is this then.

Instead of using the FC_REFLECT, because the internals of shared_bytes are not very usable through reflection.

Add two public methods to `shared_bytes`:
```c++
template <typename Stream>
inline Stream& operator<<(Stream& ds, const shared_bytes& b) {
   ds << b.size();
   ds.write(b.data(), b.size());
   return ds;
}
```
and
```c++
template <typename Stream>
inline Stream& operator>>(Stream& ds, shared_bytes& b) {
   std::size_t sz;
   ds >> sz;
   shared_bytes tmp = {sz};
   ds.read(tmp.data(), tmp.size());
   b = tmp;
   return ds;
}
```

Reply:
shared_bytes already has an out stream overload and a method that can create a shared_bytes from a hex string (which is what is streamed out).  You would just need to provide the operator for streaming in.